test/client/clf.go

```diff
@@ -20,3 +26,24 @@ func ClusterLogForwarderReady(e watch.Event) (bool, error) {
 		return false, nil
 	}
 }
+
+// ClusterLogForwarderValidationFailure expects condition type "Validation" to be set on the ClusterLogForwarder
+// resource. If no such condition can be found, it returns false, and a nil error (so that c.WaitFor can wait until
+// the condition is set, or time out if the condition is never set). If the condition is set, we expect its message
+// to match validationFailureMsg and we expect it to be "True". We also expect the "Ready" condition to be "False".
+// In that case, we return true and no error. In case of the contrary, we return false and an error.
+func ClusterLogForwarderValidationFailure(e watch.Event) (bool, error) {
+	clf := e.Object.(*loggingv1.ClusterLogForwarder)
```

Comment:
Do we need a type check here? Is it possible to receive an object that is not a CLF?

Reply:
@andreaskaris I wasn't suggesting it does not need to be cast but that seems like this is unsafe and we need to type check.  The first line just assumes it is a CLF, but isn't there a chance it could be something else?

Reply:
ah!   `e.Object.(*loggingv1.ClusterLogForwarder)`  is a go type assertion though, so IMO it's safe?
[`https://go.dev/tour/methods/](https://go.dev/tour/methods/15) "If i does not hold a T, the statement will trigger a panic. "
As in, if someone passes something else, this will panic on purpose

Are we on the same page now, or am I still misunderstanding you? Anyway, I'm still polishing some of the other code, I'll move this back from draft to non-draft when ready